libbeat/common/jsontransform/jsonhelper.go

```diff
@@ -110,3 +115,23 @@ func removeKeys(keys map[string]interface{}, names ...string) {
 		delete(keys, name)
 	}
 }
+
+func parseTimestamp(timestamp string) (time.Time, error) {
+	validFormats := []string{
+		time.RFC3339,
+		iso8601,
+	}
+
+	var err error
+	for _, f := range validFormats {
+		ts, parseErr := time.Parse(f, timestamp)
+		if parseErr != nil {
+			err = multierror.Append(err, parseErr)
```

Comment:
1. It looks like the error is never returned.
2. Maybe we should create our own error message if parsing fails instead of returning a collection.